### PR TITLE
Remove cmake warnings

### PR DESCRIPTION
**Pull request by j-rivero**
remove useless cmake code

### CODE REVIEW

**Comment by scpeters** [comment]
The warning comes from building in an isolated workspace. I think the dependency is useful in non-isolated workspaces.

I spoke to the expert @dirk-thomas who recommended that we depend on `${catkin_EXPORTED_TARGETS}` instead of `gazebo_msgs_gencpp`, as specified in the catkin documentation:
- http://docs.ros.org/api/catkin/html/howto/format2/building_msgs.html

I think that will fix the cmake warning.

File: gazebo_ros/CMakeLists.txt [hunk 1]
```diff
@@ -71,13 +71,13 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")
 
 ## Plugins
 add_library(gazebo_ros_api_plugin src/gazebo_ros_api_plugin.cpp)
-add_dependencies(gazebo_ros_api_plugin gazebo_msgs_gencpp ${PROJECT_NAME}_gencfg) # wait for gazebo_msgs to be built
+add_dependencies(gazebo_ros_api_plugin ${catkin_EXPORTED_TARGETS})
```
Comment (fmessmer) [inline gazebo_ros/CMakeLists.txt]: as this package generates dynamic_reconfigure files, you might want to actually write 
`add_dependencies(gazebo_ros_api_plugin ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})`
to make sure the respective header is there, i.e. `${${PROJECT_NAME}_EXPORTED_TARGETS}` ensures both former "_gencpp" and "_gencfg"

`

File: gazebo_ros/CMakeLists.txt [hunk 2]
```diff
@@ -71,13 +71,13 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")
...
 set_target_properties(gazebo_ros_api_plugin PROPERTIES LINK_FLAGS "${ld_flags}")
 set_target_properties(gazebo_ros_api_plugin PROPERTIES COMPILE_FLAGS "${cxx_flags}")
 target_link_libraries(gazebo_ros_api_plugin ${GAZEBO_LIBRARIES} ${SDFormat_LIBRARIES} ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${TinyXML_LIBRARIES})
 
 add_library(gazebo_ros_paths_plugin src/gazebo_ros_paths_plugin.cpp)
-add_dependencies(gazebo_ros_paths_plugin gazebo_msgs_gencpp ${PROJECT_NAME}_gencfg) # wait for gazebo_msgs to be built
+add_dependencies(gazebo_ros_paths_plugin ${catkin_EXPORTED_TARGETS})
```
Comment (fmessmer) [inline gazebo_ros/CMakeLists.txt]: same here

File: gazebo_ros/CMakeLists.txt (hunk 1 above)
Comment (scpeters) [inline gazebo_ros/CMakeLists.txt]: what do you think @j-rivero ?

Comment (j-rivero) [inline gazebo_ros/CMakeLists.txt]: > what do you think @j-rivero ?

I agree with Felix. f56f0fb826f03311ba3bc37aef4e2fefe87e01cc

**Comment by fmessmer** [comment]
+1

**Comment by j-rivero** [comment]
Thanks Felix for the comments. @scpeters ready to merge by my side.

**Comment by scpeters** [comment]
+1

**Pull request merged**